include/FffGcodeWriter.h

```diff
@@ -328,13 +328,17 @@ class FffGcodeWriter : public NoCopy
      * This adds all features (e.g. walls, skin etc.) of this \p mesh to the gcode which are printed using \p extruder_nr
      *
      * \param[in] storage where the slice data is stored.
-     * \param mesh The mesh to add to the layer plan \p gcode_layer.
+     * \param mesh_ptr The mesh to add to the layer plan \p gcode_layer.
      * \param extruder_nr The extruder for which to print all features of the mesh which should be printed with this extruder
      * \param mesh_config the line config with which to print a print feature
      * \param gcode_layer The initial planning of the gcode of the layer.
      */
-    void addMeshLayerToGCode(const SliceDataStorage& storage, const SliceMeshStorage& mesh, const size_t extruder_nr, const MeshPathConfigs& mesh_config, LayerPlan& gcode_layer)
-        const;
+    void addMeshLayerToGCode(
+        const SliceDataStorage& storage,
+        const std::shared_ptr<SliceMeshStorage>& mesh_ptr,
+        const size_t extruder_nr,
+        const MeshPathConfigs& mesh_config,
+        LayerPlan& gcode_layer) const;
```

Comment:
Are you still willing to pass the smart pointers by reference ? IMHO it makes the code less readable (and having a "reference to a pointer to an object" is bit mindfucking :smile: ). Some people advise that you create a typedef of the smart pointer so that you can give `SliceMeshStorage::Ptr` instead of `std::shared_ptr<SliceMeshStorage>` but I'm no big fan of this either..

I rely on your wisdom to do what you think is best :pray:

Reply:
It's a bit more efficient; if they don't get passed by reference, they get passed by copy. The object itself isn't very large, but the ref-counter will also be increased as well.

I don't feel very strongly about it, and probably won't even comment if I saw someone else (not) do it -- but now that it's in there, I don't feel the need to change it back.